### PR TITLE
Support wildcards in label search

### PR DESCRIPTION
**Pull request by KyleJu**
Fix https://github.com/web-platform-tests/wpt.fyi/issues/3351. Support searching wildcards in label.  

This is intended for use case https://github.com/web-platform-tests/wpt-metadata/pull/4297, and https://github.com/web-platform-tests/wpt-metadata/pull/4303

### CODE REVIEW

**Comment by KyleJu** [comment]
A screenshot for a wildcard search:
Screenshot 2023-08-11 at 9 27 35 AM (2)

File: api/query/cache/index/index_filter_test.go [hunk 1]
```diff
@@ -833,6 +833,65 @@ func TestBindExecute_TestLabel(t *testing.T) {
...
+						Test:   "/d/e/f",
+						Status: "FAIL",
+					},
+				},
+			},
+		},
+	})
+	metadata := map[string][]string{
+		"/foo/bar/b.html": {"random"},
+		"/a/*":  {"interop1", "INTEROP2"},
```
Comment (jcscottiii) [inline api/query/cache/index/index_filter_test.go:863]: There's a possible edge case. Here's the scenario: If you modify `matchingTestName` to equal `/a` and this metadata line to `"/*":  {"interop1", "INTEROP2"},`, you will get zero results as-is.

The question I have: Is it even possible for the metadata to produce this scenario (if the appropriate meta.yml was modified)? If not, you can ignore this question.

*I don't think this would be a useful scenario to even have in the metadata (since it's just a root level wildcard) but I didn't want to assume.*

**Review by jcscottiii — APPROVED** [review]
LGTM. Just one question

File: api/query/cache/index/index_filter_test.go (hunk 1 above)
Comment (KyleJu) [reply api/query/cache/index/index_filter_test.go:863]: Great question! It is a possible edge case, but it wouldn't make sense to have a root level wildcard, since it would label all WPT tests.  However, we should safeguard this case in the wpt-metadata repository.

File: api/query/cache/index/filter.go [hunk 2]
```diff
@@ -324,6 +324,18 @@ func (tl TestLabel) Filter(t TestID) bool {
 	}
 
 	labels, ok := tl.metadata[name]
+	dir := filepath.Dir(name)
+	// Dir terminates with either '.' (when the top-level is a file) or '/'
+	// (when the top-level is a directory).
+	for !ok && len(dir) > 1 {
```
Comment (KyleJu) [inline api/query/cache/index/filter.go:330]: On a second look, I want to point out a drawback of this for loop implementation. Since a test can have labels on a test level and directory level, this for loop will terminate as soon as `A` set of labels are found.  However, if a test `/a/b/c` is labelled to `foo` and `/a/*` is labelled to `bar`,  `label:bar` will not return the intended result, due to early termination at `for !ok ...`. This drawback will affect all wildcards implementation in this file.

Comment (KyleJu) [reply api/query/cache/index/filter.go:330]: I will file a bug

Comment (KyleJu) [reply api/query/cache/index/filter.go:330]: https://github.com/web-platform-tests/wpt.fyi/issues/3448
